src/s2wasm.h

```diff
@@ -370,22 +378,59 @@ class S2WasmBuilder {
     s = inputStart;
     while (*s) {
       skipWhitespace();
-      s = strstr(s, ".type");
-      if (!s) break;
-      mustMatch(".type");
-      Name name = getCommaSeparated();
-      skipComma();
-      if (!match("@function")) continue;
-      if (match(".hidden")) mustMatch(name.str);
-      mustMatch(name.str);
-      if (match(":")) {
-        info->implementedFunctions.insert(name);
-      } else if (match("=")) {
-        Name alias = getAtSeparated();
-        mustMatch("@FUNCTION");
-        info->aliasedFunctions.insert({name, alias});
+
+      // add function definitions and aliases
+      if (match(".type")) {
+        Name name = getCommaSeparated();
+        skipComma();
+        if (!match("@function")) continue;
+        if (match(".hidden")) mustMatch(name.str);
+        mustMatch(name.str);
+        if (match(":")) {
+          info->implementedFunctions.insert(name);
+        } else if (match("=")) {
+          Name alias = getAtSeparated();
+          mustMatch("@FUNCTION");
+          info->aliasedSymbols.insert({name, LinkerObject::SymbolAlias(alias, LinkerObject::Relocation::kFunction, 0, 0)});
+        } else {
+          abort_on("unknown directive");
+        }
+      // add data aliases
       } else {
-        abort_on("unknown directive");
+        Name lhs = getStrToSep();
```

Comment:
1. Yes, I've been building various programs for testing, and found that the compiler produced aliases to aliases. That code wasn't in the original version of this pull; I added it yesterday.
2. I'm not sure what you mean by matching sizes; the size of an alias should always be less than the size of the parent object minus the offset, but it doesn't necessarily have to be equal. AFAIK the aliasee is always declared before the alias, so we can check the sizes, but the current pull doesn't do this (the size field isn't used). Currently, the size information is stored separately from a `Relocation` in a `StaticObject`, but the latter are in a `staticObjects` vector that isn't efficient to search. I'm not sure if we want to change this to a map, or add a size field to relocations for this?